clients/cobol-lsp-vscode-extension/src/services/ProcessorGroups.ts

```diff
@@ -51,26 +59,58 @@ export async function loadProcessorGroupCopybookPaths(
 export async function loadProcessorGroupCopybookPathsConfig(
   item: { scopeUri: string },
   configObject: string[],
-): Promise<string[]> {
-  const cfg = await loadProcessorGroupSettings(
-    item.scopeUri,
-    "libs",
-    [] as string[],
-  );
-  const config = SettingsService.evaluateVariables(
-    [...cfg, ...configObject],
-    getVariablesFromUri(item.scopeUri, false),
-  );
+): Promise<
+  (string | ZoweDatasetConfigModel | ZoweUssConfigModel | EndevorConfigModel)[]
+> {
+  const allConfigs: (
+    | string
+    | ZoweDatasetConfigModel
+    | ZoweUssConfigModel
+    | EndevorConfigModel
+  )[] = [
+    ...(await loadProcessorGroupSettings(
+      item.scopeUri,
+      "libs",
+      [] as string[],
+    )),
+    ...configObject,
+  ];
 
+  const configs: (
+    | string
+    | ZoweDatasetConfigModel
+    | ZoweUssConfigModel
+    | EndevorConfigModel
+  )[] = [];
+  const variables = getVariablesFromUri(item.scopeUri, false);
   const wsUri = workspace.getWorkspaceFolder(Uri.parse(item.scopeUri))?.uri;
-  if (wsUri === undefined) {
-    return configObject;
+  let cleanWsFolder: string | undefined;
+  if (wsUri) cleanWsFolder = cleanWorkspaceFolderName(wsUri.fsPath);
+
+  for (const config of allConfigs) {
+    if (typeof config === "string") {
+      const tempConfig = SettingsService.evaluateVariables([config], variables);
```

Comment:
Just modify the `evaluateVariables` to take and return string only.

In the else branch then iterate over keys of the input object and adjust them all.

Reply:
Why it is needed to adjust Endevor Type ?

Reply:
Is there a particular reason why the substitutions should work only in half of the cases?

Reply:
I don't think dataset names & endevor types needs to be adjusted.